resources/scripts/ttx_diff.py

```diff
@@ -52,22 +79,30 @@
     0.1,
     "The percentage of point (glyf) or delta (gvar) values allowed to differ by one without counting as a diff",
 )
+flags.DEFINE_bool("json", False, "print results in machine-readable JSON format")
 
 
 def run(cmd: MutableSequence, working_dir: Path, log_file: str, **kwargs):
     cmd_string = " ".join(cmd)
-    print(f"  (cd {working_dir} && {cmd_string} > {log_file} 2>&1)")
-    log_file = working_dir / log_file
-    with open(log_file, "w") as log_file:
-        subprocess.run(
-            cmd,
-            text=True,
-            check=True,
-            cwd=working_dir,
-            stdout=log_file,
-            stderr=log_file,
-            **kwargs,
-        )
+    maybe_print(f"  (cd {working_dir} && {cmd_string} > {log_file} 2>&1)")
+    log_path = working_dir / log_file
+    try:
+        with open(log_path, "w") as log_file:
+            subprocess.run(
+                cmd,
+                text=True,
+                check=True,
+                cwd=working_dir,
+                stdout=log_file,
+                stderr=log_file,
+                **kwargs,
+            )
+    except subprocess.CalledProcessError as e:
+        # hack: in the JSON case I want to be able to return the stashed output
+        # in the error, so we go full python :)
```

Comment:
"full python" seems needlessly disparaging of Python.

Suggest declaring a custom error type and raising it.

Reply:
I think it's more recognizing python's unique strengths, but 🤷